hwmv2: Port all arc boards

nordicjm opened the pull request:
tests: Update board names for hwmv2
    
    Updates tests that use board names which have changed with boards v2
    

    boards: arc: hsdk4xd: Convert to v2
    
    Converts the board to hwmv2


    soc: snps_arc_hsdk4xd: Port to HWMv2
    
    Ports the snps_arc_hsdk4xd SoC configuration to hardware model
    version 2


    boards: arc: iotdk: Convert to v2
    
    Converts the board to hwmv2


    soc: snps_arc_iot: Port to HWMv2
    
    Ports the snps_arc_iot SoC configuration to hardware model version 2


    boards: arc: emsdp: Convert to v2
    
    Converts the board to hwmv2


    soc: snps_emsdp: Port to HWMv2
    
    Ports the snps_emsdp SoC configuration to hardware model version 2


    boards: arc: em_starterkit: Convert to v2
    
    Converts the board to hwmv2, documentation for this board needs to
    be vastly updated
    

    soc: snps_emsk: Port to HWMv2
    
    Ports the snps_emsk SoC configuration to hardware model version 2
    

    boards: arc: hsdk: Convert to v2
    
    Converts the board to hwmv2
    

    soc: snps_arc_hsdk: Port to HWMv2
    
    Ports the snps_arc_hsdk SoC configuration to hardware model
    version 2


    boards: arc: nsim: Convert to v2
    
    Converts the board to hwmv2


    soc: snps_nsim: Port to HWMv2
    
    Ports the snps_nsim SoC configuration to hardware model version 2


    boards: arc: qemu_arc: Convert to v2
    
    Converts the board to hwmv2
    

    soc: snps_qemu: Port to HWMv2
    
    Ports the snps_qemu SoC configuration to hardware model version 2

gmarull commented on boards/v2/qemu/qemu_arc/board.yml:
```diff
@@ -0,0 +1,10 @@
+board:
+  name: qemu_arc
+  vendor: Synopsys
```
since the board is under `qemu`, having vendor: Synopsys is a bit confusing. What is the policy with qemu boards?

nordicjm replied:
Fixed

ruuddw replied:
fyi, we had this discussion in the past. The ARC qemu is done by Synopsys, and I think not yet (at least then) upstream. That was the reason for keeping Synopsys as vendor.
But, I'm fine with moving under qemu.

tejlmand commented on boards/v2/qemu/qemu_arc/Kconfig.qemu_arc:
```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_QEMU_ARC
+	bool
```
bool not needed.

And as all board identifier are doing: `select BOARD_QEMU_ARC`, then let's move that up, and remove redundant code.
```suggestion
	select BOARD_QEMU_ARC
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/qemu/qemu_arc/Kconfig.qemu_arc:
```diff
@@ -0,0 +1,26 @@
+config BOARD_QEMU_ARC
+	bool
+
+config BOARD_QEMU_ARC_QEMU_EM4
+	bool
+	select BOARD_QEMU_ARC
+
+config BOARD_QEMU_ARC_QEMU_HS3X
+	bool
+	select BOARD_QEMU_ARC
+
+config BOARD_QEMU_ARC_QEMU_HS3X_XIP
+	bool
+	select BOARD_QEMU_ARC
+
+config BOARD_QEMU_ARC_QEMU_HS5X
+	bool
+	select BOARD_QEMU_ARC
+
+config BOARD_QEMU_ARC_QEMU_HS6X
+	bool
+	select BOARD_QEMU_ARC
```
If moving the `select` to board level, then all those lines can be removed.
```suggestion
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/nsim/Kconfig.nsim:
```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_NSIM
+	bool
```
```suggestion
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/nsim/Kconfig.nsim:
```diff
@@ -0,0 +1,84 @@
+	  and a console based on the ns16550 UART model.
+
+config BOARD_NSIM_NSIM_EM4
+	bool
```
```suggestion
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/nsim/Kconfig.nsim:
```diff
@@ -0,0 +1,84 @@
+	select BOARD_NSIM
+
+config BOARD_NSIM_NSIM_EM7D_V22
+	bool
```
```suggestion
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/nsim/Kconfig.nsim:
```diff
@@ -0,0 +1,84 @@
+	select BOARD_NSIM
+
+config BOARD_NSIM_NSIM_EM11D
+	bool
```
```suggestion
```

please cleanup following lines, even though not commented directly.

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit:
```diff
@@ -4,12 +4,23 @@
 	  EM7D of EMSK 2.3 supports secure mode.
+
+config BOARD_EM_STARTERKIT_EMSK_EM7D
+	bool
```
```suggestion
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit:
```diff
@@ -4,12 +4,23 @@
+	select SOC_EMSK_EM7D
+
+config BOARD_EM_STARTERKIT_EMSK_EM9D
+	bool
```
```suggestion
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/em_starterkit/Kconfig.em_starterkit:
```diff
@@ -4,12 +4,23 @@
+	select SOC_EMSK_EM9D
+
+config BOARD_EM_STARTERKIT_EMSK_EM11D
+	bool
```
```suggestion
```

tejlmand commented on boards/v2/synopsys/emsdp/Kconfig.emsdp:
```diff
@@ -4,12 +4,46 @@
 	  Subsystems (DFSS, SCSS, DSS) are also supported.
+
+config BOARD_EMSDP_EMSDP_EM4
+	bool
```
Please remove redundant `bool` in this file.

```suggestion
```

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/iotdk/board.yml:
```diff
@@ -0,0 +1,5 @@
+  name: iotdk
+  vendor: Synopsys
+  socs:
+  - name: arc_iot
```
newline missing.

nordicjm replied:
Fixed

tejlmand commented on boards/v2/synopsys/hsdk4xd/board.yml:
```diff
@@ -0,0 +1,5 @@
+  name: hsdk4xd
+  vendor: Synopsys
+  socs:
+  - name: arc_hsdk4xd
```
new line missing.

nordicjm replied:
Fixed

ruuddw commented on boards/v2/qemu/qemu_arc/Kconfig:
```diff
@@ -0,0 +1,23 @@
+	select QEMU_TARGET
+	select HAS_COVERAGE_SUPPORT
+
+config BOARD_QEMU_ARC_QEMU_EM4
```
I'd prefer sticking to "em" for this one, instead of "em4". This is a 'generic' board for the family, e.g. covering em6 as well. And, there are few more places (e.g. support/mdb_em.args) that have the same name.
Let's not change names as part of this restructuring. If for some reason it needs to be changed, then have a separate PR for that.

nordicjm replied:
This is the CPU, the CPU being em4, it cannot be em because there is no such soc

ruuddw replied:
The original SoC was also called EM, not sure I understand the issue with sticking with EM and HS instead of renaming all of these (globally: soc/board/docs).

nordicjm replied:
Why were two soc added with invalid names originally? em and hs are not SoCs, they are soc families, em4 and hs3x are the soc names according to https://www.synopsys.com/designware-ip/processor-solutions/arc-classic-processors/arc-hs-family.html and https://www.synopsys.com/designware-ip/processor-solutions/arc-classic-processors/arc-em-family.html

ruuddw replied:
ARC is highly configurable, instead of adding a SoC for all possible configurations we made an abstraction.
nsim_em e.g. covers both EM4 (no caches) and EM6 (with caches).
em4 and hs3x are not SoC names. These are product names for Synopsys ARC CPU IP cores (which can be configured and integrated on a SoC).

nordicjm replied:
With hwmv2, there is not the level of configurability as was present with hwmv1, so for example with hwmv1 you could configure for an stm32 board which has a soc, but you could then change the soc part number, in hwmv2 this is no longer possible, you configure and build for a specific soc, if you want a different part number, you need to alter the board files to specify that soc (if it isn't already one of the possible soc options) but you cannot change the configuration once configured. So with the em and hs socs, is that what you mean by configurability or is it something else?

ruuddw replied:
No, I mean silicon IP core configurability. That is several steps before board/soc configurability.
Let's separate hwmv2 porting and other possible improvements like changing names. My goal is to minimize changes for existing users. If you feel strongly about the names, then let's have a separate issue/PR/discussion about that.

ruuddw commented on boards/v2/qemu/qemu_arc/Kconfig:
```diff
@@ -0,0 +1,23 @@
+config BOARD_QEMU_ARC_QEMU_EM4
+	select SOC_QEMU_ARC_EM4
+
+config BOARD_QEMU_ARC_QEMU_HS3X
```
same for this one, let's keep this as generic 'hs'.

nordicjm replied:
As above

nordicjm replied:
You can't have this both ways, the soc is hs3x and it was wrong to have hs in the original board then have hs3x_xip (which is a variant), you can either have both called hs (which in my opinion is wrong because there is no hs soc) or both named hs3x.

nordicjm replied:
The options here are as follows:

- The naming in this PR is picked, em4 and hs3x, to build for the hs3x you would use `qemu_arc/qemu_hs3x` and for XIP variant `qemu_arc/qemu_hs3x/xip`
- The naming in this PR is changed so that em4 falls back to em and hs3x falls back to hs, this means the hs3x_xip name will vanish and be hs
- The existing names from hwmv1 are used, this means the whole hs3x soc has to be duplicated as hs and hs3x, this then means hs and hs3x will be different boards in qemu, it means you will have `qemu_arc/qemu_hs`, `qemu_arc/qemu_hs3x` (which will fail to configure and throw an error, it will still appear in the list of supported board names) and `qemu_arc/qemu_hs3x/xip` which will be the XIP version

nordicjm replied:
I got mixed up between different socs it seems, it has been addressed

tejlmand commented:
please rebase

nordicjm committed:
soc: snps_qemu: Port to HWMv2

Ports the snps_qemu SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: qemu_arc: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_nsim: Port to HWMv2

Ports the snps_nsim SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: nsim: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_hsdk: Port to HWMv2

Ports the snps_arc_hsdk SoC configuration to hardware model
version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: hsdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_emsk: Port to HWMv2

Ports the snps_emsk SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: em_starterkit: Convert to v2

Converts the board to hwmv2, documentation for this board needs to
be vastly updated

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: Update board names for hwmv2

Updates tests that use board names which have changed with boards v2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_emsdp: Port to HWMv2

Ports the snps_emsdp SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: emsdp: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_iot: Port to HWMv2

Ports the snps_arc_iot SoC configuration to hardware model version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: iotdk: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: snps_arc_hsdk4xd: Port to HWMv2

Ports the snps_arc_hsdk4xd SoC configuration to hardware model
version 2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: arc: hsdk4xd: Convert to v2

Converts the board to hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm commented:
@ruuddw please review

ruuddw commented:
I still see the renamings for EM and HS and still fail to understand why these are needed. Please use "EM" instead of "EM4" and "HS" instead of "HS3X". The XIP variant can be qemu_arc/qemu_hs/xip

Furthermore, do I need to do anything special to build? I tried some v2 boards, but get various errors:
warning: SPARC_CASA (defined at soc\sparc\Kconfig:4) defined without a type
warning: attempt to assign the value 'n' to the undefined symbol ARC_HAS_STACK_CHECKING
warning: attempt to assign the value 'y' to the undefined symbol ARC_MPU_ENABLE

nordicjm commented:
> I still see the renamings for EM and HS and still fail to understand why these are needed. Please use "EM" instead of "EM4" and "HS" instead of "HS3X". The XIP variant can be qemu_arc/qemu_hs/xip
> 
> Furthermore, do I need to do anything special to build? I tried some v2 boards, but get various errors: warning: SPARC_CASA (defined at soc\sparc\Kconfig:4) defined without a type warning: attempt to assign the value 'n' to the undefined symbol ARC_HAS_STACK_CHECKING warning: attempt to assign the value 'y' to the undefined symbol ARC_MPU_ENABLE

qemu_arc was renamed, can you point out exactly what boards have been renamed? That is a path issue in windows, if you rebase on `collab-hwm` it will be resolved

gmarull reviewed (dismissed):
update docs/maintainers as needed

nordicjm committed:
boards: synopsys: Add documentation index

Adds a documentation index file

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
maintainers: Update synopsys entries

Updates entries to account for hwmv2

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

ruuddw commented on boards/synopsys/em_starterkit/em_starterkit_emsk_em7d.dts:
```diff
@@ -7,7 +7,7 @@
 
 #include <mem.h>
-#include <synopsys/emsk.dtsi>
+#include <synopsys/emsk-sec.dtsi>
```
why "-sec"?

nordicjm replied:
because that is what the original board dts has?

nordicjm replied:
Fixed

ruuddw commented on boards/synopsys/em_starterkit/em_starterkit_emsk_em9d.dts:
```diff
@@ -8,7 +8,7 @@
 #include <mem.h>
 #include <synopsys/emsk.dtsi>
-#include "em_starterkit_r23.dtsi"
+#include "em_starterkit_r22.dtsi"
```
why  "_r22" instead of "_r23"?

nordicjm replied:
> The original board files were wrong, the documentation states that em11d is supported on r2.3 and r2.2 but actually did not allow selecting r2.2

With these changes, the boards can correctly be changed between r2.2 and r2.3

nordicjm replied:
Removed

ruuddw reviewed (changes requested):
Not fully finished my review, but it looks you're updating in parallel so I'll check again later.
Please let me know when to review again.

nordicjm commented on boards/synopsys/em_starterkit/em_starterkit_emsk_em11d.dts:
```diff
@@ -8,7 +8,7 @@
 #include <mem.h>
 #include <synopsys/emsk.dtsi>
-#include "em_starterkit_r23.dtsi"
+#include "em_starterkit_r22.dtsi"
```
The original board files were wrong, the documentation states that em11d is supported on r2.3 and r2.2 but actually did not allow selecting r2.2

nordicjm replied:
Removed

nordicjm commented on boards/qemu/arc/doc/index.rst:
```diff
@@ -12,7 +12,7 @@ ARCv2 and ARCv3 hardware platforms.
 
 * CPU:
-  * ARCv2 EM
+  * ARCv2 EM4
```
Fixed

nordicjm commented on boards/qemu/arc/doc/index.rst:
```diff
@@ -68,13 +68,13 @@ Programming and Debugging
 emulated environment, for example, with the :zephyr:code-sample:`synchronization` sample
-(note you may use ``qemu_arc_em``, ``qemu_arc_hs``,  ``qemu_arc_hs5x`` or
-``qemu_arc_hs6x`` depending on target CPU):
+(note you may use ``qemu_arc/qemu_em4``, ``qemu_arc/qemu_hs3x``,  ``qemu_arc/qemu_hs5x`` or
```
Fixed

nordicjm commented on boards/qemu/arc/doc/index.rst:
```diff
@@ -68,13 +68,13 @@ Programming and Debugging
    :zephyr-app: samples/synchronization
    :host-os: unix
-   :board: qemu_arc_em
+   :board: qemu_arc/qemu_em4
```
Fixed

nordicjm commented on boards/synopsys/em_starterkit/doc/index.rst:
```diff
@@ -26,18 +26,18 @@ See also this URL for details about the board:
 2.0/2.1/2.2 to 2.3 using latest firmware.
 The default configuration for EM Starter Kit boards can be found in
-:zephyr_file:`boards/arc/em_starterkit/em_starterkit_defconfig`.
+:zephyr_file:`boards/synopsis/em_starterkit/em_starterkit_defconfig`.
```
Fixed

nordicjm commented on boards/synopsys/em_starterkit/doc/index.rst:
```diff
@@ -26,18 +26,18 @@ See also this URL for details about the board:
 instruction bus and data bus. Instruction memory is called ICCM
 and data memory is called DCCM. The configuration file for EM9D
-is found in :zephyr_file:`soc/arc/snps_emsk/Kconfig.defconfig.em9d`.
+is found in :zephyr_file:`soc/synopsis/snps_emsk/Kconfig.defconfig.em9d`.
```
Fixed

nordicjm commented on boards/synopsys/em_starterkit/doc/index.rst:
```diff
@@ -26,18 +26,18 @@ See also this URL for details about the board:
-is found in :zephyr_file:`soc/arc/snps_emsk/Kconfig.defconfig.em7d` and EM11D is
-found in :zephyr_file:`soc/arc/snps_emsk/Kconfig.defconfig.em11d`.
+is found in :zephyr_file:`soc/synopsis/snps_emsk/Kconfig.defconfig.em7d` and EM11D is
+found in :zephyr_file:`soc/synopsis/snps_emsk/Kconfig.defconfig.em11d`.
```
Fixed

nordicjm commented on boards/synopsys/em_starterkit/doc/index.rst:
```diff
@@ -26,18 +26,18 @@ See also this URL for details about the board:
 access to 128KB DRAM with i-cache and d-cache. The configuration file for EM7D
-is found in :zephyr_file:`soc/arc/snps_emsk/Kconfig.defconfig.em7d` and EM11D is
-found in :zephyr_file:`soc/arc/snps_emsk/Kconfig.defconfig.em11d`.
+is found in :zephyr_file:`soc/synopsis/snps_emsk/Kconfig.defconfig.em7d` and EM11D is
```
Fixed

nordicjm commented:
@ruuddw ready for review

ruuddw commented on soc/synopsys/emsdp/Kconfig.soc:
```diff
@@ -0,0 +1,59 @@
+config SOC
+	default "emsdp_em4d" if SOC_EMSDP_EM4
+	default "emsdp_em5d" if SOC_EMSDP_EM5D
+	default "emsdp_em6d" if SOC_EMSDP_EM6
```
This looks like a typo there is no EM6D only EM6.

nordicjm replied:
Fixed

gmarull commented on boards/synopsys/index.rst:
```diff
@@ -1,7 +1,7 @@
 
-ARC Boards
-##########
+Synopsys
```
-1, all pages follow vendor name only, let’s be consistent please

nordicjm commented on boards/synopsys/nsim/board.yml:
```diff
@@ -0,0 +1,28 @@
+  vendor: Synopsys
+  socs:
+  - name: nsim_em
+  - name: nsim_em7d_v22
```
This is because you insist on em and em7d being different socs when they're really the same soc of a different revision

ruuddw reviewed (approved):
Thanks for the work! This looks ok now, we can do more testing once this gets merged and we have updated out internal regression environment for hwmv2
